### DISCUSSION TITLE
Attaching custom HTTP header to the requests sent to collector from web SDK?

### DISCUSSION THREAD

**Discussion by Ketec** [grafana/faro-web-sdk#726] [Q&A] [answered]
I looked through all the config options and did not notice any way to interact with it.
The beforeSend allows manipulating the payload.
How can i attach extra headers to all the requests it sends to the collector?1

**Comment by codecapitano** [comment, marked as answer]
Hi @Ketec 

The option is a bit hidden.

To attach custom headers you need to instantiate Faro's `FetchTrasnsport` / `OtlpHttpTRansport` 

```
initializeFaro({
    // ...
    transports: [
      new FetchTransport({
        url: 'my-collector-url/collect',
        requestOptions: {
          headers: {
            'x-foo': 'application/json',
          },
        },
      }),
    ],
   // ...
})
```
